kubelet/datadog_checks/kubelet/common.py

```diff
@@ -1,4 +1,102 @@
 # (C) Datadog, Inc. 2018
 # All rights reserved
 # Licensed under Simplified BSD License (see LICENSE)
+
+from tagger import get_tags
+
+try:
+    from container import is_excluded
+except ImportError:
+    # Don't fail on < 6.2
+    def is_excluded(name, image):
+        return False
+
 SOURCE_TYPE = 'kubelet'
+
+CADVISOR_DEFAULT_PORT = 0
+
+
+def tags_for_pod(pod_id, cardinality):
+    """
+    Queries the tagger for a given pod uid
+    :return: string array, empty if pod not found
+    """
+    return get_tags('kubernetes_pod://%s' % pod_id, cardinality)
+
+
+def tags_for_docker(cid, cardinality):
+    """
+    Queries the tagger for a given container id
+    :return: string array, empty if container not found
+    """
+    return get_tags('docker://%s' % cid, cardinality)
+
+
+def get_pod_by_uid(uid, podlist):
+    """
+    Searches for a pod uid in the podlist and returns the pod if found
+    :param uid: pod uid
+    :param podlist: podlist dict object
+    :return: pod dict object if found, None if not found
+    """
+    for pod in podlist.get("items", []):
+        try:
+            if pod["metadata"]["uid"] == uid:
+                return pod
+        except KeyError:
+            continue
+    return None
+
+
+def is_static_pending_pod(pod):
+    """
+    Return if the pod is a static pending pod
+    See https://github.com/kubernetes/kubernetes/pull/57106
+    :param pod: dict
+    :return: bool
+    """
+    try:
+        if pod["metadata"]["annotations"]["kubernetes.io/config.source"] == "api":
+            return False
+
+        pod_status = pod["status"]
+        if pod_status["phase"] != "Pending":
+            return False
+
+        return "containerStatuses" not in pod_status
+    except KeyError:
+        return False
+
+
+class ContainerFilter(object):
+    def __init__(self, podlist):
+        self.containers = {}
+
+        for pod in podlist.get('items') or []:
```

Comment:
Can we replace this by:
```python
for pod in podlist.get('items', []):
```

Reply:
Now that we have 
```python
            if self.pod_list.get("items") is None:
                # Sanitize input: if no pod are running, 'items' is a NoneObject
                self.pod_list['items'] = []
```
in `check()` we can. Before, we had a `items` key with a None value, that made the iteration fail.